sktime/transformations/series/cffilter.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+"""
+Interface to Christiano Fitzgerald asymmetric, random walk filter from `statsmodels`.
+
+Interfaces `cf_filter` from `statsmodels.tsa.filters`.
+"""
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+
+__author__ = ["ken-maeda"]
+__all__ = ["CFFilter"]
+
+
+import pandas as pd
+
+from sktime.transformations.base import BaseTransformer
+from sktime.utils.validation._dependencies import _check_soft_dependencies
+
+_check_soft_dependencies("statsmodels", severity="warning")
+
+
+class CFFilter(BaseTransformer):
+    """Filter a times series using the Christiano Fitzgerald filter.
+
+    This is a wrapper around the `cffilter` function from `statsmodels`.
+    (see `statsmodels.tsa.filters.cf_filter.cffilter`).
+
+    Parameters
+    ----------
+    low : float, optional, default = 6.0
+        Minimum period of oscillations. Features below low periodicity
+        are filtered out. For quarterly data, the default of 6 gives
+        1.5 years periodicity.
+
+    high : float, optional, default = 32.0
+        Maximum period of oscillations. Features above high periodicity
+        are filtered out. For quarterly data, the default of 32 gives
+        8 year periodicity.
+
+    drift : bool, optional, default =
+        Whether or not to substract a trend from the data.
+        The trend is estimated as np.arange(nobs)*(X[-1] -X[0])/(len(X)-1).
```

Comment:
is this correct? The `X` in `_transform` ends up being a 2D array, not a 1D array.
Also, is `nobs` the same as `len(X)`, which would be clearer?